src/support.sch

```diff
@@ -16,10 +17,46 @@
         <rule context="*">
             <assert test="
                 name()='article'
+                or name()='kwd'
+                or name()='kwd-group'
+                or name()='title'
             " role="warn">
                 &lt;<name/>&gt; is ignored.
             </assert>
         </rule>
     </pattern>
 
+    <pattern id="kwd-group-parent">
+        <rule context="kwd-group[parent::*]">
+            <let name="parent" value="name(..)"/>
+            <assert test="$parent='article-meta'" role="warn">
+                &lt;<name/>&gt; in &lt;<value-of select="$parent"/>&gt; is ignored.
+            </assert>
+        </rule>
+    </pattern>
+
+    <pattern id="kwd-group-kwd">
+        <rule context="kwd-group[parent::article-meta or not(parent::*)]">
```

Comment:
Bit annoying having to add in `not(parent::*)`, since it's purely for testing. Never valid JATS though...